source/agora/test/Flash.d

```diff
@@ -1917,3 +1918,133 @@ unittest
+    assert(charlie_diego_chan_id_res.error == ErrorCode.None,
+        charlie_diego_chan_id_res.message);
+    const charlie_diego_chan_id = charlie_diego_chan_id_res.value;
+    log.info("Charlie Diego channel ID: {}", charlie_diego_chan_id);
+    network.listener.waitUntilChannelState(charlie_diego_chan_id,
+        ChannelState.WaitingForFunding);
+
+    // await charlie & charlie channel funding transaction
+    network.expectTxExternalization(charlie_diego_chan_id);
+
+    // wait for the parties to detect the funding tx
+    charlie.waitForChannelOpen(WK.Keys.C.address, charlie_diego_chan_id);
+    diego.waitForChannelOpen(WK.Keys.D.address, charlie_diego_chan_id);
+    network.listener.waitUntilChannelState(charlie_diego_chan_id, ChannelState.Open);
+    /+++++++++++++++++++++++++++++++++++++++++++++/
+
+    /+ OPEN DIEGO => EOMER CHANNEL +/
+    /+++++++++++++++++++++++++++++++++++++++++++++/
+    // the utxo the funding tx will spend (only relevant to the funder)
+    const diego_utxo = UTXO(0, txs[2].outputs[0]);
+    const diego_utxo_hash = UTXO.getHash(hashFull(txs[2]), 0);
+    const eomer_diego_chan_id_res = diego.openNewChannel(diego_utxo, diego_utxo_hash,
+        Amount(3_000), Settle_1_Blocks, WK.Keys.E.address, false, to!string(WK.Keys.E.address));
+    assert(eomer_diego_chan_id_res.error == ErrorCode.None,
+        eomer_diego_chan_id_res.message);
+    const eomer_diego_chan_id = eomer_diego_chan_id_res.value;
+    log.info("Eomer Diego channel ID: {}", eomer_diego_chan_id);
+    network.listener.waitUntilChannelState(eomer_diego_chan_id,
+        ChannelState.WaitingForFunding);
+
+    // await charlie & charlie channel funding transaction
+    network.expectTxExternalization(eomer_diego_chan_id);
+
+    // wait for the parties to detect the funding tx
+    eomer.waitForChannelOpen(WK.Keys.E.address, eomer_diego_chan_id);
+    diego.waitForChannelOpen(WK.Keys.D.address, eomer_diego_chan_id);
+    network.listener.waitUntilChannelState(eomer_diego_chan_id, ChannelState.Open);
+    /+++++++++++++++++++++++++++++++++++++++++++++/
+
+    // also wait for all parties to discover other channels on the network
+    alice.waitForChannelDiscovery(charlie_diego_chan_id);
+    eomer.waitForChannelDiscovery(charlie_diego_chan_id);
+    alice.waitForChannelDiscovery(eomer_diego_chan_id);
+    charlie.waitForChannelDiscovery(eomer_diego_chan_id);
+    diego.waitForChannelDiscovery(alice_charlie_chan_id);
+    eomer.waitForChannelDiscovery(alice_charlie_chan_id);
+
+    // begin off-chain transactions
+    auto inv_1 = diego.createNewInvoice(WK.Keys.D.address, Amount(2_000),
+        time_t.max, "payment 1");
+
+    // This would have to use Charlie => Diego channel as a hop, but channel is private
+    alice.payInvoice(WK.Keys.A.address, inv_1.value);
+    auto inv_res = network.listener.waitUntilNotified(inv_1.value);
+    assert(inv_res == ErrorCode.PathNotFound, format("Payment %s didn't fail", inv_res));
```

Comment:
Note that nowadays, with `-checkaction=context`, you get a nice output even without the message as a parameter.